src/renderer/components/ReactFlowBox.tsx

```diff
@@ -135,19 +156,119 @@ const updateZIndexes = (
         }
     }
 
+    const nodesToAdjust = new Map<string, number>();
     // set the zIndex of all edges
     for (const e of edges) {
+        edgesById.set(e.id, e);
         let zIndex = Math.max(
-            byId.get(e.source)?.zIndex ?? STARTING_Z_INDEX,
-            byId.get(e.target)?.zIndex ?? STARTING_Z_INDEX
+            nodesById.get(e.source)?.zIndex ?? STARTING_Z_INDEX,
+            nodesById.get(e.target)?.zIndex ?? STARTING_Z_INDEX
         );
 
-        if (e.selected && zIndex < MIN_SELECTED_INDEX) {
-            zIndex += SELECTED_ADD;
+        const sourceNode = nodesById.get(e.source);
+        const targetNode = nodesById.get(e.target);
+
+        if (sourceNode && targetNode) {
+            if (e.selected && zIndex < MIN_SELECTED_INDEX) {
+                zIndex += SELECTED_ADD;
+                // If an edge is selected, make the connected nodes the same zIndex so that the handle stays above
+                if (sourceNode.type !== 'iterator') {
+                    // We need to wait until after the loop to adjust the zIndex
```

Comment:
Please don't just copy this comment everywhere. Maybe have one comment near `nodesToAdjust` explain what it does.